Mathlib/Geometry/Convex/Cone/Basic.lean

```diff
@@ -382,6 +382,96 @@ instance instAddCommSemigroup : AddCommSemigroup (ConvexCone R M) where
   add_assoc _ _ _ := SetLike.coe_injective <| add_assoc _ _ _
   add_comm _ _ := SetLike.coe_injective <| add_comm _ _
 
+section Generating
+
+/-- A convex cone `C` is generating if its linear span is the entire `R`-module `M`.
+
+`IsGenerating` is equivalent to `IsReproducing` modulo some conditions.
+See `IsReproducing.isGenerating` and `IsGenerating.isReproducing` for details. -/
+@[simp] def IsGenerating (C : ConvexCone R M) : Prop :=
+  Submodule.span R (C : Set M) = ⊤
+
+/-- The whole `R`-module `M` (viewed as the top convex cone) is generating. -/
+theorem isGenerating_top : (⊤ : ConvexCone R M).IsGenerating := by
+  simp
+
+/-- The empty convex cone is generating iff the module is a subsingleton. -/
+theorem isGenerating_bot_iff : (⊥ : ConvexCone R M).IsGenerating ↔ Subsingleton M := by
+  simpa only [IsGenerating, coe_bot, Submodule.span_empty, ← Submodule.subsingleton_iff R] using
+    subsingleton_iff_bot_eq_top
+
+/-- In a subsingleton module, the empty convex cone is generating. -/
+theorem isGenerating_bot [Subsingleton M] : (⊥ : ConvexCone R M).IsGenerating :=
+  isGenerating_bot_iff.mpr inferInstance
+
+/-- A convex cone containing a generating cone is also a generating cone. -/
+@[gcongr]
+theorem IsGenerating.mono (h : C₁ ≤ C₂) (hgen : C₁.IsGenerating) : C₂.IsGenerating := by
+  rw [IsGenerating, ← top_le_iff] at hgen ⊢
+  exact hgen.trans (Submodule.span_mono h)
+
+/-- A convex cone is reproducing if its set of element differences equals the entire module,
+i.e., every element of `M` can be written as a difference of two elements of `C`.
+
+See also (`IsGenerating`). -/
+def IsReproducing [AddCommGroup M] (C : ConvexCone R M) : Prop :=
+  (C : Set M) - (C : Set M) = Set.univ
```

Comment:
Could you do the same for this new predicate? Add a constructor, and two dot-notation lemmas...